lib/improver/utilities/cube_checker.py

```diff
@@ -171,3 +171,49 @@ def find_dimension_coordinate_mismatch(
     else:
         mismatch = list(set(second_dim_names) - set(first_dim_names))
     return mismatch
+
+
+def coord_exists_in_cube(cube, coord_name):
+    """
+    Test whether a coordinate exists within a cube, regardless of whether it
+    is a dimension, auxillary or scalar coordinate.
+
+    Args:
+        cube : iris.cube.Cube
+            The cube to be tested.
+        coord_name : str
+            The name of the coordinate to be looked for.
+
+    Returns:
+        bool
+            True if the coordinate is found within the cube, False if it is
+            not.
+
+    """
+
+    if [coord.name() for coord in cube.coords()
+            if coord_name in coord.name()]:
```

Comment:
This will match if coord_name only partial matches e.g. if coord_name = 'percentile' and within the cube there is a coord_name of 'percentile_over_realization' then this will return true. Do you really want 
`    if coord_name == coord.name()`

Reply:
Laurence corrected this.